components/common/component.js

```diff
@@ -0,0 +1,50 @@
+export class Component {
+  constructor(props = {}) {
+    this.props = props;
+    this.state = props.initialState || {};
+    this.children = props.children || [];
```

Comment:
P5;
[널 병함 연산자](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing)를 사용하면 더 안전하게 값을 다룰 수 있어요.

```suggestion
    this.state = props.initialState ?? {};
    this.children = props.children ?? [];
```